pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -232,6 +270,78 @@ func (c *K8sOrchestrator) InitTopologyServiceInController(ctx context.Context) (
 		"cluster flavor: %q", c.clusterFlavor)
 }
 
+// refreshPreferentialDatastores refreshes the preferredDatastoresMap variable
+// with latest information on which datastore is preferred in which topology domain.
+func refreshPreferentialDatastores(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	cnsCfg, err := common.GetConfig(ctx)
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to fetch CNS config. Error: %+v", err)
+	}
+	// Get VC instance.
+	vc, err := cnsvsphere.GetVirtualCenterInstance(ctx, &cnsconfig.ConfigurationInfo{Cfg: cnsCfg},
+		false)
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to get virtual center instance with error: %v", err)
+	}
+	// Get tag manager instance.
+	tagMgr, err := cnsvsphere.GetTagManager(ctx, vc)
```

Comment:
Can you logout when done with tagMgr usage?

Reply:
+1, deferred call is preferred.